### PR TITLE
fix(eslint-plugin-formatjs): make shareable configs compatible with flat config

### LINKED ISSUE #4705: [eslint-plugin] New Shareable Configs Are Not Compatible
## Which package?

eslint-plugin-formatjs

## Describe the bug

The new shareable configs (introduced by https://github.com/formatjs/formatjs/issues/4501) are in the deprecated legacy/rc format, not the new flat config format.  Most notably, the `plugins` key is an object now and not a string array.  They also need the `meta` properties.  And since the plugin as of v5 only supports eslint v9, it doesn't make sense to only support the deprecated config format. 

## To Reproduce

Use either of the new configs with eslint v9, and you'll get an error about the config format being incompatible.

### PR DESCRIPTION
**Pull request by michaelfaith**
The new shareable configs introduced by #4501 were in the deprecated legacy/rc format, not the new flat config format. Most notably, the plugins key is an object now and not a string array. They also needed the meta properties. And since the plugin as of v5 only supports eslint v9, it doesn't make sense to only support the deprecated config format.

This change addresses those issues, making the configs work (only) with the new format.

Fixes #4705

### CODE REVIEW

**Comment by longlho** [comment]
thanks for putting this together. Is there a way to write test for this?

**Comment by michaelfaith** [comment]
> thanks for putting this together. Is there a way to write test for this?

What i generally do for testing the configs, is more in the form of e2e tests. Have an "examples" folder with a set of simple projects that use the plugin and run lint as part of ci testing. Another benefit of that is it acts as a sort of documentation you can point people to for usage examples.  What do you think?

**Comment by michaelfaith** [comment]
Added `resolveJsonModule` to `tsconfig.node.json` so that the `package.json` import would work.

**Comment by longlho** [comment]
hmm ill take it from here.

File: packages/eslint-plugin-formatjs/index.ts [hunk 1]
```diff
@@ -64,119 +64,131 @@ import {
...
+        {
+          props: {
+            include: [['*', '{label,placeholder,title}']],
           },
-        ],
-        'formatjs/blocklist-elements': ['error', {elements: ['selectordinal']}],
-      },
+        },
+      ],
+      'formatjs/blocklist-elements': ['error', {elements: ['selectordinal']}],
```
Comment (michaelfaith) [inline packages/eslint-plugin-formatjs/index.ts]: Oo, the integration tests served their purpose.  Looks like there was an issue with the original config for `blocklist-elements`
```suggestion
      'formatjs/blocklist-elements': ['error', ['selectordinal']],
```
Per
image

File: packages/eslint-plugin-formatjs/index.ts [hunk 2]
```diff
@@ -64,119 +64,131 @@ import {
...
```
Comment (michaelfaith) [inline packages/eslint-plugin-formatjs/index.ts]:
```suggestion
      'formatjs/blocklist-elements': ['error', ['selectordinal']],
```

File: packages/eslint-plugin-formatjs/index.ts (hunk 1 above)
Comment (michaelfaith) [reply packages/eslint-plugin-formatjs/index.ts]: Also that screen shot made me realize that the documentation on the site should probably be updated for the new config format

**Comment by longlho** [comment]
hmm that test passes which is a little odd cause that file definitely doesn't conform to `strict`. Maybe the exit code for `pnpm` is wack

**Comment by longlho** [comment]
oh it's because it's a typescript file hmm. Now it fails which is what I expected

**Review by longlho — APPROVED** [review]

**Comment by longlho** [comment]
Thanks a lot for your contributions!

**Pull request merged**